src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.cs

```diff
@@ -771,5 +771,65 @@ public static int GetHRForLastWin32Error()
         }
 
         public static IntPtr /* IDispatch */ GetIDispatchForObject(object o) => throw new PlatformNotSupportedException();
+
+        public static void ZeroFreeBSTR(IntPtr s)
+        {
+            if (s == IntPtr.Zero)
+            {
+                return;
+            }
+            RuntimeImports.RhZeroMemory(s, (UIntPtr)SysStringByteLen(s));
+            FreeBSTR(s);
+        }
+
+        public unsafe static void ZeroFreeCoTaskMemAnsi(IntPtr s)
+        {
+            ZeroFreeCoTaskMemUTF8(s);
+        }
+
+        public static unsafe void ZeroFreeCoTaskMemUnicode(IntPtr s)
+        {
+            if (s == IntPtr.Zero)
+            {
+                return;
+            }
+            RuntimeImports.RhZeroMemory(s, (UIntPtr)(string.wcslen((char*)s) * 2));
+            FreeCoTaskMem(s);
+        }
+
+        public static unsafe void ZeroFreeCoTaskMemUTF8(IntPtr s)
+        {
+            if (s == IntPtr.Zero)
+            {
+                return;
+            }
+            RuntimeImports.RhZeroMemory(s, (UIntPtr)string.strlen((byte*)s));
+            FreeCoTaskMem(s);
+        }
+
+        public unsafe static void ZeroFreeGlobalAllocAnsi(IntPtr s)
+        {
+            if (s == IntPtr.Zero)
+            {
+                return;
+            }
+            RuntimeImports.RhZeroMemory(s, (UIntPtr)string.strlen((byte*)s));
+            FreeHGlobal(s);
+        }
+
+        public static unsafe void ZeroFreeGlobalAllocUnicode(IntPtr s)
+        {
+            if (s == IntPtr.Zero)
+            {
+                return;
+            }
+            RuntimeImports.RhZeroMemory(s, (UIntPtr)(string.wcslen((char*)s) * 2));
+            FreeHGlobal(s);
+        }
+
+        internal static uint SysStringByteLen(IntPtr s)
+        {
+            return (uint)ReadInt32(s, -4);
```

Comment:
I would not use this super-slow ReadInt32 method. These legacy methods should not be really ever used for anything.
```
return *(((uint*)s) - 1);
```

Reply:
ok, done